### PR TITLE
Setup Azure Pipelines for Mac

### PR DESCRIPTION
**Pull request by leouieda**
**Description of proposed changes**

Enables the continuous integration services from Azure Pipelines. The configuration is in `.azure-pipelines.yml` and only builds on Mac for now. This can replace the Mac builds on TravisCI to try to speed up the process a bit by having more jobs execute in parallel.

The compilation only job runs in ~7.5 min. Full tests run in ~20 min.

**Reminders**

- [ ] Make sure that your code follows our style. Use the other functions/files as a basis.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Describe changes to function behavior and arguments in a comment below the function declaration.
- [ ] If adding new functionality, add a detailed description to the documentation and/or an example.

**ToDo**:

- [x] Add badge to the README
- [x] Disable Travis Mac jobs (?)

### CODE REVIEW

**Comment by leouieda** [comment]
@GenericMappingTools/core I had to setup a "GenericMappingTools" organization in Azure as well and it doesn't sync with the Github authentication. This means that I have to add anyone who wants access to reset the builds, etc. Please let me know the email you use with a Microsoft account (or create one if you haven't) and I'll add you as admin.

**Comment by leouieda** [comment]
We can also control when the cron jobs execute so we could have this running multiple times a day if desired.

**Comment by joa-quim** [comment]
Leonardo, I'm (5 m's)  ``Quimmmmm`` @ ``hotmail.com``

**Comment by leouieda** [comment]
OK, things are working on Mac. It's faster than Travis (for Mac at least) and can serve as a good basis for setting up Windows builds. 

Should we remove the Travis Mac jobs?

**Comment by seisman** [comment]
I think we should disable travis mac builds.

File: ci/azure-pipelines-mac.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Template for the basic Mac steps in Azure Pipelines
+
+steps:
+
+- bash: |
+    set -x -e
+    brew install cmake ninja curl netcdf gdal fftw pcre2 ghostscript curl graphicsmagick || true
```
Comment (seisman) [inline ci/azure-pipelines-mac.yml]: duplicate `curl` here.

File: ci/azure-pipelines-mac.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  displayName: Install dependencies
+
+- bash: echo "##vso[task.setvariable variable=INSTALLDIR]$BUILD_SOURCESDIRECTORY/gmt-install-dir"
+  displayName: Set install location
+
+- bash: echo "##vso[task.setvariable variable=COASTLINEDIR]$INSTALLDIR/coast"
+  displayName: Set coastline location
+
+- bash: echo "##vso[task.prependpath]$INSTALLDIR/bin"
```
Comment (seisman) [inline ci/azure-pipelines-mac.yml]: duplicate `Set PATH` steps.

File: ci/azure-pipelines-mac.yml (hunk 1 above)
Comment (leouieda) [reply ci/azure-pipelines-mac.yml]: Fixed

File: ci/azure-pipelines-mac.yml (hunk 2 above)
Comment (leouieda) [reply ci/azure-pipelines-mac.yml]: Fixed

**Comment by leouieda** [comment]
Thanks for catching that @seisman

**Pull request merged**